go_covert_lib/controller/processor/symmetricEncryption/symmetricEncryption.go

```diff
@@ -0,0 +1,237 @@
+package symmetricEncryption
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/des"
+	"errors"
+)
+
+// padding is done by having 7 zero's at the start followed by the length of 
+// the original message, the contents of the message, followed by a series of
+// zero's to fill the block size requirements of the symmetric encryption 
+// algorithm 
+const padIndex = 7
+
+type SymmetricEncryption struct {
+	algorithm string
+	mode      string
+	key       []byte
+}
+
+func (c *SymmetricEncryption) Process(data []byte) ([]byte, error) {
+	data = Pad(data)
+	algorithm := c.algorithm
+
+	var (
+		blockSize int
+		block     cipher.Block
+		err       error
+	)
+
+	// based on the users choice of symmetric algorithm create a cipher
```

Comment:
Why not just put c.algorithm here in the switch statement. Why allocate the other variable?

Reply:
The code in this switch statement should be moved to the ToProcessor function, see below. Doing so would reduce code duplication and move the check on the key format to the time when the user inputs the key, instead of when they send a message.